e2e/wdio/core/tests/grid-list.e2e-spec.ts

```diff
@@ -0,0 +1,122 @@
+            expect(getElementArrayLength(moreButtonItems)).toEqual(defaultItemsQuantity + 5);
+            defaultItemsQuantity += 5;
+        }
+        click(moreButton);
+        expect(moreButton).not.toBeDisplayed();
+    });
+
+    it('Footer should be displayed and contain information', () => {
+        expect(getText(footer)).toBe(text);
+    });
+
+    it('Verify each grid list contains product counter -> product counter should be displayed for all lists', () => {
+        const arr = getElementArrayLength(gridListsArray);
+        for (let i = 0; i < arr; i++) {
+            expect(getText(gridListsTitle, i)).toContain(productTitle);
+        }
+    });
+
+    it('Verify grid list contains product counter', () => {
+        let productsQuantityFromTitle = getText(deleteModeTitle).replace(/\D/g, '');
+        const itemsArray = elementArray(gridListItemsByMode('delete'));
+        expect(productsQuantityFromTitle).toEqual(itemsArray.length.toString());
+        for (let i = 0; i < itemsArray.length; i++) {
+            scrollIntoView(deleteItemButton);
+            click(deleteItemButton);
+            productsQuantityFromTitle = getText(deleteModeTitle).replace(/\D/g, '');
+            const newArray = elementArray(gridListItemsByMode('delete'));
+            expect(productsQuantityFromTitle).toEqual(newArray.length.toString());
+        }
+    });
+
+    it(`Verify states: Text should be in bold if item is on unread state, Error message should be displayed in footer if item is on 'error' state
+    Locker button should be displayed in footer if item is on 'locked' state, Draft button should be displayed in footer if item is on 'draft' state`, () => {
+        expect(getCSSPropertyByName(unreadStateItem, fontWeight).value).toBe(bold);
+        expect(warningColor).toContain(getCSSPropertyByName(errorStateItem, color).value as string);
+        waitForClickable(lockedStateItemButton);
+        expect(getText(lockedStateItemText)).toBe(textLocked);
+    })
+
+    it('Verify selecting multiple items in "Multi select mode" component -> Multiple items can be selected. Checkbox should be checked when item is selected', () => {
+        const arrayLength = getElementArrayLength(gridListItemsByMode('multiSelect'));
+        let selectedArrayLength = getElementArrayLength(multiSelectModeSelectedItems)
+        expect(selectedArrayLength).toEqual(1);
+        for (let i = 0; i < arrayLength; i++) {
+            scrollIntoView(multiSelectModeCheckboxes, i);
+            click(multiSelectModeCheckboxes, i);
+        }
+        selectedArrayLength = getElementArrayLength(multiSelectModeSelectedItems)
+        expect(selectedArrayLength).toEqual(arrayLength - 1)
+    });
+
+    it('Verify corresponding indicator color should be displayed for all statuses', () => {
+        expect(successColor).toContain(getCSSPropertyByName(successStatusIndicator, backGroundColor).value as string);
```

Comment:
why it has a `as string`, is value is nullable?
then use `)!.value` if possible

Reply:
seems like `as string` isn't really needed here. it should return a string anyway because background-color will return with something like  "rbga(255,255,255,1)"